gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/remote/RemoteConnection.java

```diff
@@ -67,7 +67,11 @@
         // default implementation for backward compatibility to 3.2.4 - this method will probably just become
         // the new submit() in 3.3.x when the deprecation is removed
         final CompletableFuture<RemoteTraversal<?, E>> promise = new CompletableFuture<>();
-        promise.complete(submit(bytecode));
+        try {
+            promise.complete(submit(bytecode));
+        } catch (Exception t) {
```

Comment:
Chose to limit to `Exception` types instead of `Throwable` since `RuntimeException`/`Error` are probably acceptable to be thrown on calling thread since they are unexpected and likely not caused by the traversal failing.